src/main/java/seedu/task/logic/commands/ListCommand.java

```diff
@@ -1,20 +1,26 @@
 package seedu.task.logic.commands;
 
-
 /**
- * Lists all persons in the address book to the user.
+ * Abstract class to represent generic list operations.  
+ * @author xuchen
  */
-public class ListCommand extends Command {
-
-    public static final String COMMAND_WORD = "list";
-
-    public static final String MESSAGE_SUCCESS = "Listed all persons";
-
-    public ListCommand() {}
-
-    @Override
-    public CommandResult execute() {
-        model.updateFilteredListToShowAll();
-        return new CommandResult(MESSAGE_SUCCESS);
-    }
+public abstract class ListCommand extends Command {
```

Comment:
Yeah i think this is a good idea! I will change the AddCommand to a similar structure too..